task02.rb

```diff
@@ -0,0 +1,28 @@
+#2. For a given array that contains some strings output the content of those strings separated with whitespace between them, having capitalized first word and ending with ".", e.g. [[" thIs ", "Is"], nil, " My  ", ["   first"], "", :program] should output "This is my first program."
+
+def task02(arr)
+  if arr.class != Array
+    raise "The input type is invalid, expect an array"
+  end
+  if arr.empty?
+    raise "The input is empty."
+  end
+
+  format_array = arr.flatten.join(" ").strip!
+  if format_array.empty?
+    raise "Not a valid input. The input had only empty data."
+  end
+  if !format_array.scan(/[^\s\w,-]/).empty? || !format_array.scan(/_/).empty?
```

Comment:
@MaraEfimova What are those two validation actually doing? And do we need a separate one for `_`?